server/strategy.go

```diff
@@ -582,23 +582,46 @@ func decideEtcdServiceOps(apiServer *cke.Node, svc *corev1.Service) cke.Operator
 }
 
 func decideResourceOps(apiServer *cke.Node, ks cke.KubernetesClusterStatus, resources []cke.ResourceDefinition, isReady bool) (ops []cke.Operator) {
+
 	for _, res := range static.Resources {
 		// To avoid thundering herd problem. Deployments need to be created only after enough nodes become ready.
 		if res.Kind == cke.KindDeployment && !isReady {
 			continue
 		}
+		// We don't have to wait to complete resource creation or update,
 		status, ok := ks.ResourceStatuses[res.Key]
 		if !ok || res.NeedUpdate(&status) {
 			ops = append(ops, op.ResourceApplyOp(apiServer, res, !status.HasBeenSSA))
+			return ops
 		}
 	}
 	for _, res := range resources {
 		if res.Kind == cke.KindDeployment && !isReady {
 			continue
 		}
 		status, ok := ks.ResourceStatuses[res.Key]
-		if !ok || res.NeedUpdate(&status) {
+		if !ok {
 			ops = append(ops, op.ResourceApplyOp(apiServer, res, !status.HasBeenSSA))
+			// To avoid applying subsequent resources not to wait the completion the update or creation
```

Comment:
Suggestions for expression:

- This sentence is hard to read, please refrain from using "not" as possible and use words itself describing negative meaning (e.g. avoid, prevent, skip).
- It would be better to mention create and update in a chronological order.
- The word _update_ here looks like a verb. It looks difficult to use the noun form, I would like to suggest using the verb form.

```suggestion
			// Skip waiting the completion to create or update to avoid applying subsequent resources
```